### DISCUSSION TITLE
iOS 18: “Enable full trust for root certificates”

### DISCUSSION THREAD

**Discussion by armindoantunes** [mitmproxy/mitmproxy#7194] [General]
Hi All! 
Since iOS 18 beta 3 that I can't find the mitmproxy certificate in  Settings > General > About > Certificate Trust Settings. 
On beta 2 it would show at “Enable full trust for root certificates” section. But not anymore.

Has anyone find a way to make this work for iOS 18?

**Comment by iOSAppWhiz** [comment]
I also encountered the same issue after upgrading to the official version of iOS 18. Even after downloading the Apple Inc. Root certificate (https://www.apple.com/certificateauthority/), I still couldn't find it in Settings > General > About > Certificate Trust Settings. My solution was to reset all settings and content, which restored normal functionality. This seems to be a bug in iOS.

**Reply by armindoantunes** [reply]
It works! thank you! :)

**Comment by caphun** [comment]
I was able to get this to work without resetting my device. 

First look for the `mitmproxy-ca-cert.pem` file on the host machine, for me that was in the `~/.mitmproxy` folder. Then find a way to open this file in Safari on your iPhone and download it to your device. Now, the mitmproxy profile should appear in Settings > General > VPN & Device Management. Install it, then go to Settings > General > About > Certificate Trust Settings and enable it.

**Reply by HQSINGLE** [reply]
What does host machine mean?

**Reply by llama** [reply]
@caphun that sounds like the normal process. Were you ever encountering the issue? It seems to be sporadic, only affecting some users.

**Reply by caphun** [reply]
> What does host machine mean?

the machine that's running mitmproxy

**Reply by caphun** [reply]
> @caphun that sounds like the normal process. Were you ever encountering the issue? It seems to be sporadic, only affecting some users.

Yes, I had the same issue as OP after upgrading my device to ios 18. It's only affecting devices that already had the mitmproxy certificate installed prior to the OS update.

**Comment by dreampiggy** [comment]
Does the above solution means, I need to "Grab my previous installed cert pem on my iPhone", the  re-install it again, and it can be appeared on iPhone's CA Trust Settings ?


Now I seems need to use iMazing to dump my iPhone backup and extract the cert pem to have a try

**Comment by nebukazar911** [comment]
Just chiming in... I have the exact same issue. From what I've read, this seems to be a recognized bug currently being discussed at https://developer.apple.com/forums/thread/764673 and will hopefully be fixed in version 18.2.

**Reply by nebukazar911** [reply]
I have just updated to 18.2 and the issue appears to be fixed. I can now see the self-signed CA cert under Certificate Trust Settings.

**Reply by mhils** [reply]
Thank you for following up, @nebukazar911! 🍰
